### ISSUE TITLE
istio 0.2.4 new ingress does not work until ingress controller killed

### ISSUE THREAD

**Issue by robertfreund** [istio/issues#79]
#On istio 0.2.4 , adding new ingress results in a 404 until the ingress pod is killed.

minikube 0.22.2
kubectl version
``` json
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.6", GitCommit:"4bc5e7f9a6c25dc4c03d4d656f2cefd21540e28c", GitTreeState:"clean", BuildDate:"2017-09-14T06:55:55Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.5", GitCommit:"17d7182a7ccbb167074be7a87f0a68bd00d58d97", GitTreeState:"clean", BuildDate:"2017-09-18T20:30:29Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```
Add a ingress:
``` yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: client-ingress
  annotations:
    kubernetes.io/ingress.class: istio
    ingress.kubernetes.io/rewrite-target: /
spec:
  rules:
  - http:
      paths:
      - path: /client
        backend:
          serviceName: client
          servicePort: 8000
      - path: /client/.*
        backend:
          serviceName: client
          servicePort: 8000
```
``` 
kubectl create -f <(istioctl kube-inject -f client.yml)
ingress "client-ingress" created
service "client" created
deployment "client" created
```
```
$  curl -v 192.168.99.100:31670/client/metrics
*   Trying 192.168.99.100...
* TCP_NODELAY set
* Connected to 192.168.99.100 (192.168.99.100) port 31670 (#0)
> GET /client/metrics HTTP/1.1
> Host: 192.168.99.100:31670
> User-Agent: curl/7.50.3
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< date: Wed, 27 Sep 2017 21:31:56 GMT
< server: envoy
< content-length: 0
< x-envoy-upstream-service-time: 0
< 
* Curl_http_done: called premature == 0
* Connection #0 to host 192.168.99.100 left intact
```

Sometimes the connections fail with 404 sometimes they succeed. It is very sporadic. 
```
[2017-09-27T21:31:56.096Z] "GET /client/metrics HTTP/1.1" 404 - 0 0 3 0 "172.17.0.1" "curl/7.50.3" "df6cdf06-f445-91c6-b19b-fac292d30f72" "192.168.99.100:31670" "172.17.0.17:8080"
[libprotobuf INFO external/mixerclient_git/src/quota_cache.cc:232] Add a new Referenced for quota cache: RequestCount, reference: Absence-keys: Exact-keys: context.protocol, 
[2017-09-27T21:32:30.027Z] "GET /client/metrics HTTP/1.1" 200 - 0 6178 11 8 "172.17.0.1" "curl/7.50.3" "f07dc4d9-1169-9d2e-941a-45b3e1df7db9" "192.168.99.100:31670" "172.17.0.17:8080"
```

When I kill ingress controller it seems 100% of time it works fine with no changes.
```
kubectl delete pods -l istio=ingress --namespace=istio-system
```

**Comment by andraxylia** [comment]
It may be related to the refreshRate for Envoy, which is currently set to 30secs.

**Comment by andraxylia** [comment]
Could be related to istio/pilot#1325

**Comment by linsun** [comment]
yea this sounds related to istio/pilot#1325 which there is a temp fix for 0.2:   istio/istio#958

**Comment by andraxylia** [comment]
@robertfreund could you please try again with the fix for istio/istio#958 and the new istio.yaml from master?

**Comment by robertfreund** [comment]
updated with istio.yaml from master. no issues with 404. thanks!
